crates/ruff_cli/src/commands/format.rs

```diff
@@ -145,24 +146,111 @@ fn format_path(
     source_type: PySourceType,
     mode: FormatMode,
 ) -> Result<FormatCommandResult, FormatCommandError> {
-    let unformatted = std::fs::read_to_string(path)
-        .map_err(|err| FormatCommandError::Read(Some(path.to_path_buf()), err))?;
-
-    let options = settings.to_format_options(source_type, &unformatted);
-    debug!("Formatting {} with {:?}", path.display(), options);
-
-    let formatted = format_module_source(&unformatted, options)
-        .map_err(|err| FormatCommandError::FormatModule(Some(path.to_path_buf()), err))?;
+    // Extract the sources from the file.
+    let source_kind = match SourceKind::from_path(path, source_type) {
+        Ok(Some(source_kind)) => source_kind,
+        Ok(None) => return Ok(FormatCommandResult::Unchanged),
```

Comment:
Isn't that an error condition?

Reply:
No, it's (e.g.) a non-Python notebook. (This is the same as in .)